crates/http_client/src/http_client.rs

```diff
@@ -48,12 +67,12 @@ pub trait HttpClient: 'static + Send + Sync {
         req: http::Request<AsyncBody>,
     ) -> BoxFuture<'static, anyhow::Result<Response<AsyncBody>>>;
 
-    fn get<'a>(
-        &'a self,
+    fn get(
+        &self,
         uri: &str,
         body: AsyncBody,
         follow_redirects: bool,
-    ) -> BoxFuture<'a, anyhow::Result<Response<AsyncBody>>> {
+    ) -> BoxFuture<'static, anyhow::Result<Response<AsyncBody>>> {
```

Comment:
What's the motivation for removing the lifetime bound? AFAICT most callers of this method will `.await` it immediately anyways, and I worry this might mess up drop ordering / allow invalid code if some logic somewhere assumes the client outlives its response futures (which appears somewhat likely)